packages/core/lib/console.js

```diff
@@ -130,6 +133,14 @@ class Console extends EventEmitter {
     abstractions.forEach(abstraction => {
       contextVars[abstraction.contract_name] = abstraction;
     });
+
+    // make sure the repl gets the new contracts in its context
+    if (this.repl) {
```

Comment:
This guard can be removed if the repl creation is done before getting accounts in the `Console.start` method.

```
      this.repl = repl.start(...);
      this.interfaceAdapter.getAccounts().then(fetchedAccounts ...)
```

Reply:
Good catch! I reworked this part a bit to incorporate this suggestion.